### ISSUE TITLE
Crash on navigation in dev mode in 'next' branch

### ISSUE THREAD

**Issue by petecarapetyan** [modernweb-dev/rocket#375]
`next` branch

This rocket-03-minimal instance with dozen or so nested .md files crashes on navigation quite reliably, at least in development mode.

https://github.com/petecarapetyan/rocket_nav_iteration

Guessing it probably wouldn't crash in deployed production but will test separately.

To replicate:
- create a `rocket-03-minimal` using the usual `npx @rocket/create@latest` command
- add a nested set of folders and .md files as might be reflected by normal usage. (one could copy the folders from the above linked project)
- or alternately just run `npm start` from the above linked project

Result:
- you can navigate to each of the nested files maybe 8 or so times?
- then it barfs with an error such as below

```
FATAL ERROR: v8::FromJust Maybe value is Nothing.
 1: 0x10e548815 node::Abort() (.cold.1) [/Users/petecarapetyan/.volta/tools/image/node/16.13.2/bin/node]
 2: 0x10d247b09 node::Abort() [/Users/petecarapetyan/.volta/tools/image/node/16.13.2/bin/node]
 3: 0x10d247c7f node::OnFatalError(char const*, char const*) [/Users/petecarapetyan/.volta/tools/image/node/16.13.2/bin/node]
``` 
and the shell simply exits

**Comment by petecarapetyan** [comment]
As expected, does not happen on a production site, only in dev mode

**Comment by GreenJimmy** [comment]
I am getting it too, picked rocket-01-hydration-starter from npx, did an npm install then start and it wont start at all

**Comment by daKmoR** [comment]
thanks for letting me know how you encountered the issue... I could now finally reproduce it consistently even via a test 🎉

Seems a complex one but having a failing test is half the battle - so off to the debug corner it is - wish me luck 😬

**Comment by daKmoR** [comment]
ah yes, that was an interesting one... two web workers writing to the same file at the same time in parallel 🙈
let's hope something like this never returns 😅

It's fixed now via `@rocket/engine@0.2.4` and I tested it with the hydration starter - which works fine now 🎉

thanks again for reporting this - if there are any more bugs be sure to report them... it's bug hunting season 😬
